src/bin/pg_autoctl/cli_do_fsm.c

```diff
@@ -399,3 +425,124 @@ cli_do_fsm_step(int argc, char **argv)
 	}
 	fformat(stdout, "%s ➜ %s\n", oldRole, newRole);
 }
+
+
+/*
+ * cli_do_fsm_get_nodes displays the list of nodes parsed from the nodes file
+ * on-disk. A nodes file is only used when running with --disable-monitor.
+ */
+static void
+cli_do_fsm_get_nodes(int argc, char **argv)
+{
+	Keeper keeper = { 0 };
+	KeeperConfig *config = &(keeper.config);
+
+	bool missingPgdataIsOk = true;
+	bool pgIsNotRunningIsOk = true;
+	bool monitorDisabledIsOk = true;
+
+	*config = keeperOptions;
+
+	if (!keeper_config_read_file(config,
+								 missingPgdataIsOk,
+								 pgIsNotRunningIsOk,
+								 monitorDisabledIsOk))
+	{
+		/* errors have already been logged */
+		exit(EXIT_CODE_BAD_CONFIG);
+	}
+
+	if (!config->monitorDisabled)
+	{
+		log_fatal("The monitor is not disabled, there's no nodes file");
+		exit(EXIT_CODE_BAD_CONFIG);
+	}
+
+	if (!keeper_read_nodes_from_file(&keeper, &(keeper.otherNodes)))
+	{
+		/* errors have already been logged */
+		exit(EXIT_CODE_INTERNAL_ERROR);
+	}
+
+	(void) printNodeArray(&(keeper.otherNodes));
+}
+
+
+/*
+ * cli_do_fsm_set_nodes parses the list of nodes parsed from the nodes file
+ * on-disk. A JSON array of nodes objects is expected. A nodes file is only
+ * used when running with --disable-monitor.
+ */
+static void
+cli_do_fsm_set_nodes(int argc, char **argv)
```

Comment:
Future improvement: I guess it would be nice if this could read from STDIN.

Reply:
Yeah, that's a good point. That's for another day though, I think.